### ISSUE TITLE
Support `<amp-video src="...">` for video cache

### ISSUE THREAD

**Issue by mszylkowski** [ampproject/amphtml#33896]
When using a video that is specified with `<amp-video src>`, `amp-video` will use that source directly with the underlying `video` element, not allowing the video cache to use the cached sources.

Fix: When using the video cache, it should use the `src` attribute to retrieve the cache videos but it should not play the origin video.

cc @ampproject/wg-stories

**Comment by gmajoulet** [comment]
I think the fix would be to, within `amp-video`:
1. Remove all `source` elements, if any, as the browser behavior is to ignore them (cf: https://html.spec.whatwg.org/#concept-media-load-algorithm)
2. Propagate the `src` of the HTMLMediaElement on a newly created HTMLSourceElement
3. Remove the `src` of the HTMLMediaElement (otherwise the Story media pool will ignore the HTMLSourceElements)
4. fetch/add/sort the cached video sources

The Story media pool will then handle everything correctly.

**Comment by alanorozco** [comment]
Could you please provide a snippet that exemplifies this the problem? As I understood by using `amp-orig-src` we already are treating this correctly.

**Comment by mszylkowski** [comment]
We're adding a functionality where videos that specify the `cache` attribute (eg: `cache="google"`) will be able to use a cache on origin, which requires the component to fetch the cache video urls (check #33466 for this impl). Currently it fetches the video urls and adds them to the `video` element, but if `amp-video` has `src` specified, the children `source` tags will be ignored, so publishers videos wouldn't be able to use the cached videos if video source is specified this way.

TLDR: This ticket will solve the problem for `<amp-video cache="*" src="*">`, which is a different usecase than `<amp-video><source src="*" amp-orig-src="*">`

**Comment by gmajoulet** [comment]
Closed by https://github.com/ampproject/amphtml/pull/34570
